pkg/combustion/custom.go

```diff
@@ -41,30 +41,52 @@ func configureCustomFiles(ctx *image.Context) ([]string, error) {
 
 func handleCustomFiles(ctx *image.Context) error {
 	fullFilesDir := generateComponentPath(ctx, filepath.Join(customDir, customFilesDir))
-	_, err := copyCustomFiles(fullFilesDir, ctx.CombustionDir, nil)
+	err := copyCustomFiles(fullFilesDir, ctx.CombustionDir)
 	return err
 }
 
 func handleCustomScripts(ctx *image.Context) ([]string, error) {
 	fullScriptsDir := generateComponentPath(ctx, filepath.Join(customDir, customScriptsDir))
 	executablePerms := fileio.ExecutablePerms
-	scripts, err := copyCustomFiles(fullScriptsDir, ctx.CombustionDir, &executablePerms)
+	scripts, err := copyCustomScripts(fullScriptsDir, ctx.CombustionDir, &executablePerms)
 	return scripts, err
 }
 
-func copyCustomFiles(fromDir, toDir string, filePermissions *os.FileMode) ([]string, error) {
+func copyCustomFiles(fromDir, toDir string) error {
+	if _, err := os.Stat(fromDir); os.IsNotExist(err) {
+		return nil
+	}
+
+	dirEntries, err := os.ReadDir(fromDir)
+	if err != nil {
+		return fmt.Errorf("reading the custom files directory at %s: %w", fromDir, err)
+	}
+
+	// If the directory exists but there's nothing in it, consider it an error case
```

Comment:
I don't think we should consider this an error case. It's possible, however incredibly unlikely, that there would be a reason the user would want an empty directory structure (e.g. they are going to run a script that generates something). It's a stretch, but in this case -- as compared to the image config directory sub-directories -- I don't see a huge value in this check.

Reply:
I don't mind this, previously this was here when we would only handle files so I kept it but I can see the situation where they want to maintain the subdirectory structure

Reply:
Scratch that, this check should remain because it checks to make sure the `custom/files` directory itself isn't empty. This aligns with our practice where if any "top" level directory is present but empty we handle it as an error

Reply:
Gotcha, that works. Eventually I want to potentially revisit that policy across the board, but better to stay consistent for now.